maths/armstrong_numbers.py

```diff
@@ -42,6 +45,45 @@ def armstrong_number(n: int) -> bool:
         temp //= 10
     return n == sum
 
+def plurperfect_number(n: int) -> bool:
+    """Return True if n is a plurperfect number or False if it is not
+    >>> plurperfect_number(153)
+    True
+    >>> plurperfect_number(200)
+    False
+    >>> plurperfect_number(1634)
+    True
+    >>> plurperfect_number(0)
+    False
+    >>> plurperfect_number(-1)
+    False
+    >>> plurperfect_number(1.2)
+    False
+    >>> plurperfect_number(24678051)
+    True
+    >>> plurperfect_number(115132219018763992565095597973971522401)
+    True
+    """
+
+    if not isinstance(n, int) or n < 1:
+        return False
+
+    # Init a "histogram" of the digits
+    digit_histogram = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]

```

Comment:
Do not change the code for this one but just so we know the list comprehension is possible...
```python
digit_histogram = [str(n).count(str(i)) for i in range(10)]
```

Reply:
I was thinking in something more "algorithmical" and using less specific language constructs. Even because a "heavy Python" solution using conversion to strings and list operations already exists.

(Further, if I am not wrong, it looks like this code runs through the number 10 times, one for each digit. whereas the one I submitted runs the number only once.)